### ISSUE TITLE
Dagster Embedded - Detected conflicting node definitions with the same name "sync"

### ISSUE THREAD

**Issue by picinatto** [dagster-io/dagster#17208]
### Dagster version

dagster, version 1.5.3

### What's the issue?

When creating more than one asset using sling (more tables to sync) get the same name error

File: `elt -> assets -> cameras -> __init__.py`

```
cameras_alertas = build_sling_asset(
    asset_spec=AssetSpec(key="cameras_alertas", group_name="cameras"),
    source_stream="public.alertas",
    target_object="raw_cameras_alertas",
    mode=SlingMode.INCREMENTAL,
    primary_key="id_alerta",
)

cameras_motoristas = build_sling_asset(
    asset_spec=AssetSpec(key="cameras_motoristas", group_name="cameras"),
    source_stream="public.motoristas",
    target_object="raw_cameras_motoristas",
    mode=SlingMode.INCREMENTAL,
    primary_key="id_motorista",
)
```

File: `elt -> __init__.py`

```
from dagster import Definitions, load_assets_from_package_module

from .resources import sling
from .assets import cameras


cameras_alertas = load_assets_from_package_module(
    cameras,
)


defs = Definitions(
    assets=[*cameras_alertas],
    resources={
        "sling": sling,
    },
)
```

When I run `dagster dev` I get the following error:

`dagster._core.errors.DagsterInvalidDefinitionError: Detected conflicting node definitions with the same name "sync"`

If I comment `cameras_motoristas` and leave only one asset definition the code runs normally

### What did you expect to happen?

Be able to create more than one sling asset without error

### How to reproduce?

Try duplicating the following code from the tutorial on [https://docs.dagster.io/integrations/embedded-elt#examples]

```
# pyright: reportGeneralTypeIssues=none
# pyright: reportOptionalMemberAccess=none

import os

from dagster_embedded_elt.sling import (
    SlingMode,
    SlingResource,
    SlingSourceConnection,
    SlingTargetConnection,
    build_sling_asset,
)

from dagster import AssetSpec

source = SlingSourceConnection(
    type="postgres",
    host="localhost",
    port=5432,
    database="my_database",
    user="my_user",
    password=os.getenv("PG_PASS"),
)

target = SlingTargetConnection(
    type="snowflake",
    host="hostname.snowflake",
    user="username",
    database="database",
    password=os.getenv("SF_PASSWORD"),
    role="role",
)

sling = SlingResource(source_connection=source, target_connection=target)

asset_def = build_sling_asset(
    asset_spec=AssetSpec("my_asset_name"),
    source_stream="public.my_table",
    target_object="marts.my_table",
    mode=SlingMode.INCREMENTAL,
    primary_key="id",
)

asset_def2 = build_sling_asset(
    asset_spec=AssetSpec("my_asset_name2"),
    source_stream="public.my_table2",
    target_object="marts.my_table2",
    mode=SlingMode.INCREMENTAL,
    primary_key="id",
)
```

### Deployment type

Local

### Deployment details

python 3.11.2

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by PedramNavid** [comment]
Thanks for reporting this @picinatto ! This is resolved with #17283!

**Comment by picinatto** [comment]
Thanks you @PedramNavid! Awesome work!
